src/utils/compileWithFragment.js

```diff
@@ -1,6 +1,15 @@
 // @flow
 import gqlTag from 'graphql-tag';
 import serialize from 'babel-literal-to-ast';
+import template from '@babel/template';
+
+const uniqueFn = template.ast(`
```

Comment:
Basically we append this function inside a `.reduce` call after the `.concat` we use to append the fragments.